### PR TITLE
Jest!

### PR DESCRIPTION
**Pull request by giladgray**
replace mocha/chai with jest for testing. wow so good.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,10 +38,21 @@
     "demo": "./cli.js 'test/fixtures/**/*'",
     "lint": "tslint --project .",
     "lint.fix": "tslint --fix --project . ",
-    "test": "mocha --compilers ts:ts-node/register 'test/**/*Tests.ts'",
+    "test": "jest",
     "tsc": "tsc --project . && echo 'OK'",
     "watch": "tsc --project . --watch"
   },
+  "jest": {
```
Comment (adidahiya) [inline package.json]: IMO this clutters package.json too much, please move this to its own jest.config.json file

Comment (giladgray) [inline package.json]: ah sure i was just following the guide

Comment (giladgray) [inline package.json]: http://facebook.github.io/jest/docs/getting-started.html#using-typescript

**Comment by adidahiya** [comment]
Can you also refactor the test files so that they live within src and use the `__tests__/` folder structure convention? That way you don't have to maintain multiple copies of the folder structure

**Comment by giladgray** [comment]
`src/__tests__/plugin.test.ts`?

**Comment by giladgray** [comment]
The current tests don't use the fixtures, they mock the files using strings. I'm still trying to figure out how to best test the plugins.

**Pull request merged**